extensions/standard-processors/modbus/FetchModbusTcp.h

```diff
@@ -0,0 +1,143 @@
+      .supportsExpressionLanguage(true)
+      .build();
+  EXTENSIONAPI static constexpr auto SSLContextService = core::PropertyDefinitionBuilder<>::createProperty("SSL Context Service")
+      .withDescription("The Controller Service to use in order to obtain an SSL Context. If this property is set, messages will be sent over a secure connection.")
+      .isRequired(false)
+      .withAllowedTypes<minifi::controllers::SSLContextService>()
+      .build();
+  EXTENSIONAPI static constexpr auto RecordSetWriter = core::PropertyDefinitionBuilder<>::createProperty("Record Set Writer")
+    .withDescription("Specifies the Controller Service to use for writing results to a FlowFile. ")
+    .isRequired(false)
+    .withAllowedTypes<core::RecordSetWriter>()
+    .build();
+
+  EXTENSIONAPI static constexpr auto Properties = std::array<core::PropertyReference, 8>{
+    Hostname,
+    Port,
+    UnitIdentifier,
+    IdleConnectionExpiration,
+    ConnectionPerFlowFile,
+    Timeout,
+    SSLContextService,
+    RecordSetWriter
+  };
+
+  EXTENSIONAPI static constexpr auto Success = core::RelationshipDefinition{"success", "Successfully processed"};
+  EXTENSIONAPI static constexpr auto Failure = core::RelationshipDefinition{"failure", "An error occurred processing"};
+  EXTENSIONAPI static constexpr auto Relationships = std::array{Success, Failure};
+
+  EXTENSIONAPI static constexpr bool SupportsDynamicProperties = true;
+  EXTENSIONAPI static constexpr bool SupportsDynamicRelationships = false;
+  EXTENSIONAPI static constexpr auto InputRequirement = core::annotation::Input::INPUT_ALLOWED;
+  EXTENSIONAPI static constexpr bool IsSingleThreaded = true;
+
+  ADD_COMMON_VIRTUAL_FUNCTIONS_FOR_PROCESSORS
+
+  void onSchedule(core::ProcessContext& context, core::ProcessSessionFactory& session_factory) override;
+  void onTrigger(core::ProcessContext& context, core::ProcessSession& session) override;
+  void initialize() override;
+
+ private:
+  void readDynamicPropertyKeys(const core::ProcessContext& context);
+  void processFlowFile(const std::shared_ptr<utils::net::ConnectionHandlerBase>& connection_handler,
+    core::ProcessContext& context,
+    core::ProcessSession& session,
+    const std::shared_ptr<core::FlowFile>& flow_file);
+
+  auto readModbus(const std::shared_ptr<utils::net::ConnectionHandlerBase>& connection_handler,
+      const std::unordered_map<std::string, std::unique_ptr<ReadModbusFunction>>& address_map) -> nonstd::expected<core::Record, std::error_code>;
+  auto sendRequestsAndReadResponses(utils::net::ConnectionHandlerBase& connection_handler,
+      const std::unordered_map<std::string, std::unique_ptr<ReadModbusFunction>>& address_map) -> asio::awaitable<nonstd::expected<core::Record, std::error_code>>;
+  auto sendRequestAndReadResponse(utils::net::ConnectionHandlerBase& connection_handler,
+      const ReadModbusFunction& read_modbus_function) -> asio::awaitable<nonstd::expected<core::RecordField, std::error_code>>;
```

Comment:
Why trailing return types? In these cases, they don't depend on parameter identifiers, so I'd keep them in front for consistency.
```suggestion
  nonstd::expected<core::Record, std::error_code> readModbus(const std::shared_ptr<utils::net::ConnectionHandlerBase>& connection_handler,
      const std::unordered_map<std::string, std::unique_ptr<ReadModbusFunction>>& address_map);
  asio::awaitable<nonstd::expected<core::Record, std::error_code>> sendRequestsAndReadResponses(utils::net::ConnectionHandlerBase& connection_handler,
      const std::unordered_map<std::string, std::unique_ptr<ReadModbusFunction>>& address_map);
  asio::awaitable<nonstd::expected<core::RecordField, std::error_code>> sendRequestAndReadResponse(utils::net::ConnectionHandlerBase& connection_handler,
      const ReadModbusFunction& read_modbus_function);
```